include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +513,58 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class RandomAccessIterator>
+auto mode(RandomAccessIterator first, RandomAccessIterator last)
+{
+    using Real = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
+
+    std::vector<Real> sorted(first, last);
+    std::sort(sorted.begin(), sorted.end());
+
+    std::vector<Real> modes {};
+    Size max_counter {};
```

Comment:
Are you sure `max_counter` will be initialized to zero? There might be a custom type that doesn't do that.